connectors/sources/servicenow.py

```diff
@@ -327,6 +434,111 @@ async def ping(self):
             logger.exception("Error while connecting to the ServiceNow.")
             raise
 
+    def _format_doc(self, data):
+        """Format document for handling empty values & type casting.
+
+        Args:
+            data (dict): Fetched record from ServiceNow.
+
+        Returns:
+            dict: Formatted document.
+        """
+
+        data = {key: value for key, value in data.items() if value}
+        data.update(
+            {
+                "_id": data["sys_id"],
+                "_timestamp": iso_utc(parser.parse(data["sys_updated_on"])),
+            }
+        )
+        return self.serialize(doc=data)
+
+    async def _attachment_fetcher(self, table_sys_id):
+        """Method to add attachment metadata to queue
+
+        Args:
+            table_sys_id (str): Table Id for fetching attachment metadata.
+        """
+
+        async for attachment_metadata in self.servicenow_client.fetch_data(
+            url=ENDPOINTS["ATTACHMENT"], params={"table_sys_id": table_sys_id}
+        ):
+            formatted_attachment_metadata = await asyncio.to_thread(
```

Comment:
question: Do you think it's worth to spawn a separate thread for updating a dict vs. simply running it in a "blocking" way? What's the rationale behind using a separate thread here?

Reply:
Yeah! We can remove overhead for spawning separate thread